server/src/main/scala/org/http4s/server/middleware/AutoSlash.scala

```diff
@@ -16,12 +15,13 @@ object AutoSlash {
   def apply[F[_]](service: HttpService[F])(implicit F: Monad[F]): HttpService[F] = Service.lift {
     req =>
       service(req).flatMap {
-        case _: Pass[F] =>
+        case Pass() =>
           val p = req.uri.path
           if (p.isEmpty || p.charAt(p.length - 1) != '/')
-            F.pure(Pass[F])
+            Pass.pure[F]
           else {
-            val withSlash = req.withUri(req.uri.copy(path = p.substring(0, p.length - 1)))
+            val caret = req.attributes(Request.Keys.PathInfoCaret)
+            val withSlash = req.withUri(req.uri.copy(path = p.substring(caret, p.length - 1)))
```

Comment:
Doesn't this lose the scriptName part of the path?  Should this be `req.withPathInfo`?

I'll bet this is also broken on older versions.

Reply:
Maybe, I don't know how `pathInfo` works.

The problem right now is that the prefixed part is back in the URI when it's passed to a service that doesn't expect it.

I guess it should `get` and `getOrElse` on the attribute as well.